network/transport/v2/protocol.go

```diff
@@ -214,19 +212,27 @@ func (p *protocol) handlePrivateTxRetryErr(hash hash.SHA256Hash) error {
 		return nil
 	}
 
-	conn := p.connectionList.Get(grpc.ByConnected(), grpc.ByNodeDID(senderDID))
-
-	if conn == nil {
-		return fmt.Errorf("unable to retrieve payload, no connection found (tx=%s, DID=%s)", hash.String(), senderDID)
+	// Broadcast query to all TX participants we've got a connection to
+	sent := false
+	for _, curr := range pal {
+		conn := p.connectionList.Get(grpc.ByConnected(), grpc.ByNodeDID(curr))
+		if conn != nil {
+			err = conn.Send(p, &Envelope{Message: &Envelope_TransactionPayloadQuery{
+				TransactionPayloadQuery: &TransactionPayloadQuery{
+					TransactionRef: tx.Ref().Slice(),
+				},
+			}})
+
+			if err != nil {
+				log.Logger().Warnf("Failed to send TransactionPayloadQuery message to private TX participant (tx=%s, PAL=%v): %v", hash.String(), pal, err)
+			} else {
+				sent = true
+			}
+		}
 	}
 
-	err = conn.Send(p, &Envelope{Message: &Envelope_TransactionPayloadQuery{
-		TransactionPayloadQuery: &TransactionPayloadQuery{
-			TransactionRef: tx.Ref().Slice(),
-		},
-	}})
-	if err != nil {
-		return fmt.Errorf("failed to send TransactionPayloadQuery message(tx=%s, DID=%s): %w", hash.String(), senderDID, err)
+	if !sent {
+		return fmt.Errorf("unable to retrieve private TX payload, no connections to any of the participants (tx=%s, PAL=%v)", hash.String(), pal)
 	}
 
```

Comment:
Could be removed I think as there is already a log entry for when send fails

Reply:
The error return is logged as error, so the error formatting here is ok. The first part of the sentence is a bit of a duplicate. The wrapper adds `retry of TransactionPayloadQuery failed: %v`